kernel/src/mem.rs

```diff
@@ -68,8 +68,8 @@ impl<L, T> AppSlice<L, T> {
         self.len
     }
 
-    pub unsafe fn ptr(&self) -> *const T {
-        self.ptr.ptr.get() as *const T
+    pub fn ptr(&self) -> *const T {
```

Comment:
yes, and in fact you should be able to do this without using the `unsafe` keyword at all:

```rust
pub fn ptr(&self) -> *mut T {
    *self.ptr.ptr
}
```

![screenshot](https://cloud.githubusercontent.com/assets/4689/25915130/75cc9f94-358e-11e7-8b59-c62b965380a7.png)